### PR TITLE
Array initialization fixes for C#

### PR DESCRIPTION
**Pull request by codeFather2**

### CODE REVIEW

File: Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs [hunk 1]
```diff
@@ -488,10 +488,21 @@ public override Ust VisitObjectCreationExpression(ObjectCreationExpressionSyntax
...
+            var textSpan = node.GetTextSpan();
+
+            if (node.Kind() == SyntaxKind.ArrayInitializerExpression)
+            {
+                var sizes = new List<Expression> { new IntLiteral(children.Length) };
+                result = new ArrayCreationExpression(new TypeToken("", textSpan), sizes, children.ToList(), textSpan);
+            }
+            else
+            {
+                result = new MultichildExpression(children, node.GetTextSpan());
```
Comment (gabriel-fallen) [inline Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs]: Maybe `result = new MultichildExpression(children, textSpan);` for brevity?

File: Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs [hunk 2]
```diff
@@ -488,10 +488,21 @@ public override Ust VisitObjectCreationExpression(ObjectCreationExpressionSyntax
...
+            Ust result = null;
             var children = node.Expressions.Select(e => (Expression)VisitAndReturnNullIfError(e))
                 .ToArray();
-            
-            var result = new MultichildExpression(children, node.GetTextSpan());
+            var textSpan = node.GetTextSpan();
+
+            if (node.Kind() == SyntaxKind.ArrayInitializerExpression)
+            {
+                var sizes = new List<Expression> { new IntLiteral(children.Length) };
```
Comment (gabriel-fallen) [inline Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs]: 2D, 3D and so on array initializers are impossible?

Comment (codeFather2) [reply Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs]: In this kind of initializer i decide to represent 2d as 1d of 1d arrays and etc. I do this in order not to collect information from child arrays

File: Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs [hunk 3]
```diff
@@ -488,10 +488,21 @@ public override Ust VisitObjectCreationExpression(ObjectCreationExpressionSyntax
...
+                result = new ArrayCreationExpression(new TypeToken("", textSpan), sizes, children.ToList(), textSpan);
+            }
+            else
+            {
```
Comment (KvanTTT) [inline Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs]: What other alternatives are available here? Object constructor?

Comment (codeFather2) [reply Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs]: Initializers not currently supported. Dictionary initializer for example

**Pull request merged**